app/api/routes.py

```diff
@@ -480,16 +517,21 @@ def create_resource(json, db):
         db.session.commit()
         index.save_object(new_resource.serialize_algolia_search)
 
-    except (AlgoliaUnreachableHostException, AlgoliaException) as e:
-        logger.exception(e)
-        print(f"Algolia failed to index new resource '{new_resource.name}'")
-
     except IntegrityError as e:
         logger.exception(e)
         return utils.standardize_response(status_code=422)
 
+    except (AlgoliaUnreachableHostException, AlgoliaException) as e:
+        if (environ.get("FLASK_ENV") != 'development'):
+            db.session.rollback()
```

Comment:
Does this work as expected? Specifically does it rollback the committed resources? From searching around it seems its unlikely to do such a thing

Reply:
We could just use `db.session.add(new_resource)` and after all the created resources return error free, then we commit. But I could be wrong so I would love to know more.

Reply:
Or have you yet to implement the slack suggestions?

Reply:
Yes, I forgot that you had helped me solve the logic issue associated with this problem. I'll go ahead and update it with your suggestions from slack and you can review that. Thanks for having my back on this